packages/http-client-java/generator/http-client-generator-test/CadlRanch-Tests.ps1

```diff
@@ -3,15 +3,16 @@
 $ErrorActionPreference = 'Stop'
 Set-StrictMode -Version 3.0
 
-try {
-    Write-Host "Running cadl ranch tests"
-    Write-Host "Starting the test server"
-    npm run testserver-start
-    mvn clean test --no-transfer-progress -T 1C
-    Write-Host "Stopping the test server"
-    npm run testserver-stop
-    
-}
-finally {
-    Write-Host "Finished running the cadl ranch tests"
+Write-Host "Running cadl ranch tests"
+
+Write-Host "Starting the test server"
+npm run testserver-start
+Write-Host "Compile and run the tests"
+mvn clean test --no-transfer-progress -T 1C
+if ($LASTEXITCODE -ne 0) {
+  exit $LASTEXITCODE
 }
```

Comment:
Appears I need to explicitly call exit on failure.

Otherwise the pipeline is pass https://dev.azure.com/azure-sdk/public/_build/results?buildId=4163857&view=logs&j=d9775086-896f-5f20-fea4-657a2b9c6aca&t=cfe335ea-f007-53bd-28c8-29bf189d3721&l=1360